src/boot/bootctl-uki.c

```
@@ -0,0 +1,96 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "bootctl.h"
+#include "bootctl-uki.h"
+#include "fd-util.h"
+#include "parse-util.h"
+#include "pe-header.h"
+
+#define DOS_FILE_MAGIC "MZ"
+#define PE_FILE_MAGIC  "PE\0\0"
+
+static const uint8_t name_osrel[8] = ".osrel";
+static const uint8_t name_linux[8] = ".linux";
+static const uint8_t name_initrd[8] = ".initrd";
+
+static struct PeSectionHeader* pe_sections(FILE *uki, uint16_t *n)
+{
+        struct PeSectionHeader *sections;
+        struct DosFileHeader dos;
+        struct PeHeader pe;
+        uint16_t scount;
+        off_t soff;
+        size_t rc;
+
+        rc = fread(&dos, 1, sizeof(dos), uki);
+        if (rc != sizeof(dos)) {
+                log_error("DOS header read error");
+                return NULL;
+        }
```

please propagate the error to match our usual coding style. i.e. change to:

```c
        rc = fread(&dos, 1, sizeof(dos), uki);
        if (rc != sizeof(dos)) {
                if (ferror(uk))
                        return log_error_errno(errno, "Failed to read DOS header: %m");
                goto not_found;
        }
        …
not_found:
        *ret = NULL;
        *ret_n = 0;
        return 0;
```